src/buffer/out/textBuffer.cpp

```diff
@@ -3184,32 +3185,46 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
 
 // Searches through the entire (committed) text buffer for `needle` and returns the coordinates in absolute coordinates.
 // The end coordinates of the returned ranges are considered inclusive.
-std::vector<til::point_span> TextBuffer::SearchText(const std::wstring_view& needle, bool caseInsensitive) const
+bool TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags, std::vector<til::point_span>& results) const
 {
-    return SearchText(needle, caseInsensitive, 0, til::CoordTypeMax);
+    return SearchText(needle, flags, 0, til::CoordTypeMax, results);
 }
 
 // Searches through the given rows [rowBeg,rowEnd) for `needle` and returns the coordinates in absolute coordinates.
 // While the end coordinates of the returned ranges are considered inclusive, the [rowBeg,rowEnd) range is half-open.
-std::vector<til::point_span> TextBuffer::SearchText(const std::wstring_view& needle, bool caseInsensitive, til::CoordType rowBeg, til::CoordType rowEnd) const
+bool TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags, til::CoordType rowBeg, til::CoordType rowEnd, std::vector<til::point_span>& results) const

```

Comment:
nit: Instead of using `results` as an out parameter you could consider returnining a `std::optional<std::vector<...>>`, similar in spirit to a `std::expected` in the future (C++23).
It could then be split up here via something like
```cpp
auto results = textBuffer.SearchText(needle, _flags, _results);
_ok = results.has_value();
_results = std::move(results).value_or({}); // std::optional has a && (move) overload
```

This would also potentially work well in other places like so:
```cpp
const auto results = textBuffer.SearchText(needle, _flags, _results).value_or({});
```